From 1be65ea1aa1b27ca08d6799d67da97c368d4a66a Mon Sep 17 00:00:00 2001
From: Chris Burr <christopher.burr@cern.ch>
Date: Fri, 23 Aug 2019 12:09:13 +0200
Subject: [PATCH] Replace isinstance str with six.string_types

---
 Core/Workflow/Workflow.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Core/Workflow/Workflow.py b/Core/Workflow/Workflow.py
index c35c2a61b5..76f4c418f0 100755
--- a/Core/Workflow/Workflow.py
+++ b/Core/Workflow/Workflow.py
@@ -4,6 +4,7 @@
 
 import os, re, types
 import xml.sax
+import six
 from DIRAC.Core.Workflow.Parameter import *
 from DIRAC.Core.Workflow.Module import *
 from DIRAC.Core.Workflow.Step import *
@@ -34,7 +35,7 @@ def __init__( self, obj = None, name = None ):
 
     elif isinstance( obj, Workflow ):
       self.fromWorkflow( obj )
-    elif isinstance( obj, str ):
+    elif isinstance( obj, six.string_types ):
       self.parameters = ParameterCollection( None )
       self.step_instances = InstancesPool( self )
       self.step_definitions = DefinitionsPool( self )
